File: cloud/dao/region.go
package dao

import "github.com/abulo/ratel/v3/stores/null"

// Region 地区表 region
type Region struct {
	Id         *int64        `db:"id,-" json:"id"`                //bigint 区域编号,PRI
	Name       *string       `db:"name" json:"name"`              //varchar 区域名称
	ParentId   *int64        `db:"parent_id" json:"parentId"`     //bigint 父级编号
	Status     *int32        `db:"status" json:"status"`          //tinyint 状态;
	Deleted    *int32        `db:"deleted" json:"deleted"`        //tinyint 是否删除;
	Sort       *int32        `db:"sort" json:"sort"`              //int 排序;
	Creator    null.String   `db:"creator" json:"creator"`        //varchar 创建人
	CreateTime null.DateTime `db:"create_time" json:"createTime"` //datetime 创建时间
	Updater    null.String   `db:"updater" json:"updater"`        //varchar 更新人
	UpdateTime null.DateTime `db:"update_time" json:"updateTime"` //datetime 更新时间
	Children   []*Region     `db:"-" json:"children"`             // 子级
}

File: admin/src/api/interface/region.ts
export namespace Region {
  // 接口请求数据
  export interface ReqRegionList {
    name?: string; //区域名称
    deleted?: number; //是否删除
    status?: number; // 状态;
    tree?: number; // 是否树形结构
  }
  // 单个数据接口
  export interface ResRegionItem {
    id: number; // 区域编号
    name: string; // 区域名称
    parentId: number; // 父级编号
    status: number; // 状态;
    deleted: number; // 是否删除;
    sort: number; // 排序;
  }
  // 返回数据
  export interface ResRegionList extends ResRegionItem {
    children?: ResRegionList[]; // 子菜单
  }
}
